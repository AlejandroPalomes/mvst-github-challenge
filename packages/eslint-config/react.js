const { resolve } = require("node:path");

const project = resolve(process.cwd(), "tsconfig.json");

/*
 * This is a custom ESLint configuration for use a library
 * that utilizes React.
 *
 * This config extends the Vercel Engineering Style Guide.
 * For more information, see https://github.com/vercel/style-guide
 *
 */

module.exports = {
  extends: [
    "@vercel/style-guide/eslint/browser",
    "@vercel/style-guide/eslint/typescript",
    "@vercel/style-guide/eslint/react",
  ].map(require.resolve),
  parserOptions: {
    project,
  },
  plugins: ["only-warn"],
  globals: {
    JSX: true,
  },
  settings: {
    "import/resolver": {
      typescript: {
        project,
      },
    },
  },
  ignorePatterns: ["node_modules/", "dist/", ".eslintrc.js", "**/*.css"],
  // add rules configurations here
  rules: {
    "import/no-default-export": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "jsx-a11y/click-events-have-key-events": "off",
    "no-console": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "react/jsx-no-leaked-render": "off",
    "jsx-a11y/no-static-element-interactions": "off",
    "react/function-component-definition": [2, { "namedComponents": "arrow-function" }]
  },
};
