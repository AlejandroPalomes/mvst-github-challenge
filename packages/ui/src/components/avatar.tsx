import { type FC } from 'react';

const AvatarSizes = ['s', 'm', 'l'];
type AvatarSize = typeof AvatarSizes[number];

export interface InputProps {
  src: string;
  size?: AvatarSize;
}

export const Avatar: FC<InputProps> = ({ src, size}) => {
  
  return (
    <img
      alt="Avatar"
      className="logo"
      src={src}
    />
  );
};
