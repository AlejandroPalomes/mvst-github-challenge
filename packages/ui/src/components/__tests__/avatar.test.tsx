import { render, screen } from "@testing-library/react";
import { Avatar } from "../avatar";

describe('Avatar test', () => {
  it('should render without crashing', () => {
    render(<Avatar src="https://i.pravatar.cc/150?img=5"/>);
    expect(screen.getByAltText(/Avatar/)).toBeInTheDocument();
  });

  it('should render with the correct classes for "s" size', () => {
    const renderedComponent = render(<Avatar size="s" src="https://i.pravatar.cc/150?img=5"/>);
    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('should render with the correct classes for "l" size', () => {
    const renderedComponent = render(<Avatar size="l" src="https://i.pravatar.cc/150?img=5"/>);
    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('should render with the correct classes for "xl" size', () => {
    const renderedComponent = render(<Avatar size="xl" src="https://i.pravatar.cc/150?img=5"/>);
    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
})