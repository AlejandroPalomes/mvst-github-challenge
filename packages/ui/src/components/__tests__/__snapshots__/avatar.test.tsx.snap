// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar test > should render with the correct classes for "l" size 1`] = `
<body>
  <div>
    <img
      alt="Avatar"
      class="rounded-full w-20 h-20"
      src="https://i.pravatar.cc/150?img=5"
    />
  </div>
</body>
`;

exports[`Avatar test > should render with the correct classes for "s" size 1`] = `
<body>
  <div>
    <img
      alt="Avatar"
      class="rounded-full w-6 h-6"
      src="https://i.pravatar.cc/150?img=5"
    />
  </div>
</body>
`;

exports[`Avatar test > should render with the correct classes for "xl" size 1`] = `
<body>
  <div>
    <img
      alt="Avatar"
      class="rounded-full w-32 h-32"
      src="https://i.pravatar.cc/150?img=5"
    />
  </div>
</body>
`;
